src/GoogleAdWordsEventForwarder.js

```diff
@@ -70,51 +71,151 @@
+                'send-to': gtagSiteId + '/' + conversionLabel
+            };
+
+            var customProps = getCustomProps(mPEvent, isPageEvent);
+
+            var payload = Object.assign({}, conversionPayload, customProps);
+
+            return payload;
+        }
+
+        function generateGtagCommerceEvent(mPEvent, conversionLabel, isPageEvent) {
+            var conversionPayload = {
+                'send-to': gtagSiteId + '/' + conversionLabel
+            };
+
+            var customProps = getCustomProps(mPEvent, isPageEvent);
+
+            if (mPEvent.ProductAction.ProductActionType === mParticle.ProductActionType.Purchase
+                && mPEvent.ProductAction.TransactionId) {
                 if (event.ProductAction.ProductActionType === mParticle.ProductActionType.Purchase
                     && event.ProductAction.TransactionId) {
                     adWordEvent.google_conversion_order_id = event.ProductAction.TransactionId;
                 }
+                conversionPayload.order_id = mPEvent.ProductAction.TransactionId;
+            }
 
-                if (event.CurrencyCode) {
-                    adWordEvent.google_conversion_currency = event.CurrencyCode;
-                }
+            if (mPEvent.CurrencyCode) {
+                conversionPayload.currency = mPEvent.CurrencyCode;
+            }
 
-                if (event.ProductAction.TotalAmount) {
-                    adWordEvent.google_conversion_value = event.ProductAction.TotalAmount;
-                }
+            if (mPEvent.ProductAction.TotalAmount) {
+                conversionPayload.value = mPEvent.ProductAction.TotalAmount;
+            }
+
+            var payload = Object.assign({}, conversionPayload, customProps);
+
+            return payload;
+        }
 
-                adWordEvent.google_custom_params = getCustomProps(event, isPageEvent);
+        // Sends final event to Google or queues if Google isn't ready
+        function sendOrQueueEvent(conversionPayload) {
+            if (window.gtag && forwarderSettings.enableGtag == 'True') {
+                gtag('event', 'conversion', conversionPayload);
+            } else if (window.google_trackConversion) {
+                window.google_trackConversion(conversionPayload);
+            } else {
+                eventQueue.push(conversionPayload);
+            }
+        }
 
-                sendOrQueueEvent(adWordEvent);
+        function logCommerce(event, isPageEvent) {
+            var isPageEvent = false;
+            var conversionLabel = getConversionLabel(event, isPageEvent);
 
-                return true;
+            if (typeof (conversionLabel) !== 'string') {
+                return false;
             }
 
-            return false;
+            var eventPayload = generateCommerceEvent(event, conversionLabel, isPageEvent);
+
+            if (!eventPayload) {
+                return false;
+            }
+
+            sendOrQueueEvent(eventPayload);
```

Comment:
Note to self: We probably don't need to queue within the log event functions b/c that should happen during initialization check in `processEvent`